File: java-context-os/src/main/java/com/owencli/contextos/pipeline/middleware/BuildMiddleware.java
package com.owencli.contextos.pipeline.middleware;

import com.owencli.contextos.builder.ContextBuilder;
import com.owencli.contextos.pipeline.PipelineContext;
import com.owencli.contextos.pipeline.PipelineMiddleware;
import java.util.concurrent.CompletableFuture;

public class BuildMiddleware implements PipelineMiddleware {
    private final ContextBuilder b;
    public BuildMiddleware(ContextBuilder cb) { this.b = cb; }
    public String name() { return "build"; }
    public int order() { return 300; }
    public CompletableFuture<Void> execute(PipelineContext ctx) {
        return b.build(ctx.taskSpec()).thenAccept(ctx::setUnifiedContext);
    }
}

File: context_os/pipeline/middlewares/build_middleware.py
"""BuildMiddleware — 上下文构建。"""

from context_os.builder.builder import ContextBuilder
from context_os.pipeline.context import PipelineContext
from context_os.pipeline.middleware import PipelineMiddleware


class BuildMiddleware(PipelineMiddleware):
    """收集身份/对话/环境/记忆 → UnifiedContext。"""

    def __init__(self, builder: ContextBuilder):
        self._b = builder

    def name(self) -> str:
        return "build"

    def order(self) -> int:
        return 300

    async def execute(self, ctx: PipelineContext) -> None:
        ctx.unified_context = await self._b.build(ctx.task_spec)
